stdlib/list.mli

```diff
@@ -142,6 +142,11 @@ val fold_right : ('a -> 'b -> 'b) -> 'a list -> 'b -> 'b
 (** [List.fold_right f [a1; ...; an] b] is
    [f a1 (f a2 (... (f an b) ...))].  Not tail-recursive. *)
 
+val concat_map : ('a -> 'b list) -> 'a list -> 'b list
+(** [List.concat_map f l] is [List.concat (List.map f l)]. *)
+
+val flat_map : ('a -> 'b list) -> 'a list -> 'b list
```

Comment:
Why provide to names for the same operation ?

Reply:
(I mean if you have to provide more than one name I'd rather have `bind` and `join` with the right interface)

Reply:
Just because we provide two names for `concat` (`flatten`) neither of which is deprecated and both `concat_map` and `flat_map` are common names for this operation in other libraries/languages.

Reply:
I don't find these compelling arguments. By multiplying the names you are just increasing the cognitive burden of code readers.

Reply:
I don't particularly disagree. Although I think I slightly prefer to have these line up with `concat`/`flatten` and then separately have a conversation about deprecating one of those names. (I vote for getting rid of `flatten`, and `Seq.flat_map` whilst we're at it).

Reply:
I use `flatten` regularly because I find it more visual than `concat`, and I wouldn't be very happy if my code started breaking because the function is removed. I'm not sure what is the cost with having two names exposed for a new function, if both are good names (and I think `concat_map` and `flat_map` are both okay), but I would also be fine with having just one of the two.

Reply:
> I'm not sure what is the cost with having two names

Choice always has a cost both for writers and readers. 

I can already see the day when someone will want to add `List.join` to systematize with `Option` and `Result`'s interfaces and we will have three names for the same function.

Reply:
I believe https://github.com/ocaml/ocaml/pull/8760 has deprecated this discussion.